src/tags/object/Ranker.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * libraries
+ */
+import React from 'react';
+import { inject, observer } from 'mobx-react';
+import { flow, types } from 'mobx-state-tree';
+
+import Registry from '../../core/Registry';
+import { AnnotationMixin } from '../../mixins/AnnotationMixin';
+import ProcessAttrsMixin from '../../mixins/ProcessAttrs';
+import Base from './Base';
+import { parseValue } from '../../utils/data';
+
+/**
+ * components
+ */
+import Ranker from '../../components/Ranker/Ranker';
+import { transformData } from '../../components/Ranker/createData';
+
+/**
+ * @name Ranker
+ * @meta_title Ranker Tag displays items that can be dragged and dropped between columns
+ * @meta_description Customize Label Studio by displaying and sorting results for machine learning and data science projects.
+ * @param {string} value Data field value containing JSON type for Ranker
+ * @param {string} [valueType] Value to define the data type in Ranker
+ */
+const Model = types
+  .model({
+    type: 'ranker',
+    value: types.maybeNull(types.string),
+    _value: types.frozen([]),
+    valuetype: types.optional(types.string, 'json'),
+    /**
+     * rank: 1 column, reorder to rank
+     * select: 2 columns, drag results to second column to select
+     */
+    mode: types.optional(types.enumeration(['rank', 'select']), 'select'),
+    title: types.optional(types.string, ''),
+
+  })
+  .views(self => ({
+    get dataSource() {
+      const data = self._value;
+      const result = self.result.value.ranker;
+      let columns = [];
+
+      if (self.mode === 'rank') {
+        columns = [result.map(id => data.find(d => d.id === id))];
+      }
+      else {
+        columns = [
+          data.filter(d => !result.includes(d.id)),
+          result.map(id => data.find(d => d.id === id)),
+        ];
+      }
+      //grabs data from the createData file
+      return transformData(columns, self.title.split('|'));
+    },
+    get result() {
+      return self.annotation.results.find(r => r.from_name === self);
+    },
+    get resultType() {
```

Comment:
add comment here that because this tag is not control tag, but behaves like the one, we have to emulate `type`s logic.

Reply:
```suggestion
    // because this tag is not a control tag, but behaves like the one,
    // we have to emulate `type`s logic to generate results.
    get resultType() {
```